helper/testhelpers/testhelpers.go

```diff
@@ -262,215 +271,273 @@ func PassthroughWithLocalPathsFactory(ctx context.Context, c *logical.BackendCon
 
-	localopts.Logger = logger.Named("perf-sec-dr")
-	localopts.FirstCoreNumber += numCores
-	ret.PerfSecondaryDRCluster, _ = ConfClusterAndCore(t, conf, &localopts)
+	localopts.Logger = opts.Logger.Named("perf-pri-dr")
+	localopts.FirstCoreNumber += len(ret.PerfSecondaryCluster.Cores)
+	ret.PerfPrimaryDRCluster = MakeClusterAndStorage(t, InmemBackendSetup, conf, &localopts)
 
-	builder := &ReplicatedTestClustersBuilder{clusters: ret}
-	builder.setupFourClusterReplication(t)
+	localopts.Logger = opts.Logger.Named("perf-sec-dr")
+	localopts.FirstCoreNumber += len(ret.PerfPrimaryDRCluster.Cores)
+	ret.PerfSecondaryDRCluster = MakeClusterAndStorage(t, InmemBackendSetup, conf, &localopts)
 
-	// Wait until poison pills have been read
-	time.Sleep(45 * time.Second)
-	EnsureCoresUnsealed(t, ret.PerfPrimaryCluster)
-	EnsureCoresUnsealed(t, ret.PerfSecondaryCluster)
-	EnsureCoresUnsealed(t, ret.PerfPrimaryDRCluster)
-	EnsureCoresUnsealed(t, ret.PerfSecondaryDRCluster)
+	SetupFourClusterReplication(t, ret.PerfPrimaryCluster, ret.PerfSecondaryCluster, ret.PerfPrimaryDRCluster, ret.PerfSecondaryDRCluster)
 
 	return ret
 }
 
-type ReplicatedTestClustersBuilder struct {
-	clusters               *ReplicatedTestClusters
-	perfToken              string
-	drToken                string
-	perfSecondaryRootToken string
-	perfSecondaryDRToken   string
+func (r *ReplicatedTestClusters) SetupTwoClusterPerfReplication(t testing.T, maskSecondaryToken bool) {
+	SetupTwoClusterPerfReplication(t, r.PerfPrimaryCluster, r.PerfSecondaryCluster, maskSecondaryToken)
 }
 
-func SetupTwoClusterPerfReplication(t testing.T, pri, sec *vault.TestCluster) {
-	clusters := &ReplicatedTestClusters{
-		PerfPrimaryCluster:   pri,
-		PerfSecondaryCluster: sec,
+func SetupTwoClusterPerfReplication(t testing.T, pri, sec *vault.TestCluster, maskSecondaryToken bool) {
+	EnablePerfPrimary(t, pri)
+
+	var publicKey string
+	if maskSecondaryToken {
+		publicKey = generatePublicKey(t, sec)
 	}
-	builder := &ReplicatedTestClustersBuilder{clusters: clusters}
-	builder.setupTwoClusterReplication(t)
+	perfToken := GetPerformanceToken(t, pri, sec.ID, publicKey)
+
+	EnablePerformanceSecondary(t, perfToken, pri, sec, false, false)
 }
 
-func (r *ReplicatedTestClustersBuilder) setupTwoClusterReplication(t testing.T) {
-	t.Log("enabling perf primary")
-	r.enablePerfPrimary(t)
-	WaitForActiveNode(t, r.clusters.PerfPrimaryCluster)
-	r.getPerformanceToken(t)
-	t.Log("enabling perf secondary")
-	r.enablePerformanceSecondary(t)
+func GetDRReplicatedClusters(t testing.T, setup SetupGetter, conf *vault.CoreConfig, opts *vault.TestClusterOptions) *ReplicatedTestClusters {
+	clusters := PrepDRReplicatedClusters(t, setup, conf, opts)
+	SetupTwoClusterDRReplication(t, clusters.PerfPrimaryCluster, clusters.PerfPrimaryDRCluster, false)
+	return clusters
+}
+
+func (r *ReplicatedTestClusters) AddDRSecondaryCluster(t testing.T, setup SetupGetter, conf *vault.CoreConfig, opts *vault.TestClusterOptions) {
```

Comment:
Do we need another helper that adds a DR secondary to a PerfSecondary?

Reply:
Not for my current refactorings.  I'm actually unsure if I want to keep ReplicatedTestClusters as is, but I needed to stop and get my work merged before going further, this has already been a very protracted refactoring.